File: game_client/Assets/Scripts/ClientProtocol/Common/TreasureHuntEnum/ETreasureHuntGainType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.TreasureHuntEnum
{

/// <summary>
/// 太空寻宝获得类型
/// </summary>
public enum ETreasureHuntGainType {
	NONE, //0 ==== 
	[InspectorName("ORE - idx[1] - 矿石")]
	ORE, //1 ==== 矿石
	[InspectorName("TREASURE - idx[2] - 奇物")]
	TREASURE, //2 ==== 奇物
	[InspectorName("REWARD - idx[3] - 奖励")]
	REWARD, //3 ==== 奖励
}

public class ETreasureHuntGainTypeComparer : IEqualityComparer<ETreasureHuntGainType>{
	public bool Equals(ETreasureHuntGainType x, ETreasureHuntGainType y) { return x == y; }
	public int GetHashCode(ETreasureHuntGainType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 4;
}
}

File: game_server/ServerProtocol/src/Common/TreasureHuntEnum/ETreasureHuntGainType.java
package Common.TreasureHuntEnum;

/*********
 * 太空寻宝获得类型
 **/
public enum ETreasureHuntGainType {
	NONE, //0 ==== 
	ORE, //1 ==== 矿石
	TREASURE, //2 ==== 奇物
	REWARD, //3 ==== 奖励
	;
public static final ETreasureHuntGainType[]  ETreasureHuntGainType_Values = ETreasureHuntGainType.values();
public static final int ETreasureHuntGainType_Length = ETreasureHuntGainType_Values.length;
public static ETreasureHuntGainType ETreasureHuntGainType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= ETreasureHuntGainType_Length){ return null; }
	return ETreasureHuntGainType_Values[_ivalue];
}
}

